src/Shared/OpportunisticIntern.cs

```diff
@@ -465,6 +465,71 @@ internal StringInternTarget(string target)
             public bool ReferenceEquals(string other) => ReferenceEquals(_target, other);
         }
 
+        /// <summary>
+        /// Wrapper over a substring of a string.
+        /// </summary>
+        internal struct SubstringInternTarget : IInternable
```

Comment:
Isn't this basically just a span?

Reply:
Yes, `IInternable` is conceptually equivalent to `ReadOnlySpan<char>`.  It's not straightforward to switch, though, because we also provide an `IInternable` implementation on top of `StringBuilder` and `StringBuilder` contents cannot be represented with a single span. There's this [new method](https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.getchunks) so we would have to make `OpportunisticIntern` take a collection of spans which would very likely mean an allocation. Or maintain two copies of the code, one for span, one for SB.

Also, as Rainer mentioned, there's likely not much gain to be realized on .NET Framework.

I'll think about it some more and make measurements. Thank you!